mlfinlab/labeling/fixed_time_horizon.py

```diff
@@ -1,67 +1,74 @@
-    :param look_forward: (int) Number of ticks to look forward when calculating future return rate. (1 by default)
-                        If n is the numerical value of look_forward, the last n observations will return a label of NaN
-                        due to lack of data to calculate the forward return in those cases.
+    :param prices: (pd.Series or pd.DataFrame) Time-indexed stock prices used to calculate returns.
+    :param threshold: (float or pd.Series) When the absolute value of return exceeds the threshold, the observation is
+                    labeled with 1 or -1, depending on the sign of the return. If return is less, it's labeled as 0.
+                    Can be dynamic if threshold is inputted as a pd.Series, and threshold.index must match prices.index.
+                    If resampling is used, the index of threshold must match the index of prices after resampling.
+                    If threshold is negative, then the directionality of the labels will be reversed. If no threshold
+                    is provided, it is assumed to be 0 and the sign of the return is returned.
+    :param resample_by: (str) If not None, the resampling period for price data prior to calculating returns. 'B' = per
+                        business day, 'W' = week, 'M' = month, etc. Will take the last observation for each period.
+                        For full details see `here.
+                        <https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#dateoffset-objects>`_
+    :param lag: (bool) If True, returns will be lagged to make them forward-looking.
     :param standardized: (bool) Whether returns are scaled by mean and standard deviation.
     :param window: (int) If standardized is True, the rolling window period for calculating the mean and standard
                     deviation of returns.
-    :return: (pd.Series) -1, 0, or 1 denoting whether return for each tick is under/between/greater than the threshold.
-                    The final look_forward number of observations will be labeled np.nan. Index is same as index of
-                    close.
+    :return: (pd.Series or pd.DataFrame) -1, 0, or 1 denoting whether the return for each observation is
+                    less/between/greater than the threshold at each corresponding time index.
     """
-    # Calculate returns
-    forward_return = close.pct_change(periods=look_forward).shift(-look_forward)
+    # Apply resample period, if applicable.
+    if resample_by is not None:
+        prices = prices.resample(resample_by).last()
 
-    # Warning if look_forward is greater than the length of the series.
-    if look_forward >= len(forward_return):
-        warnings.warn('look_forward period is greater than the length of the Series. All labels will be NaN.',
-                      UserWarning)
+    # Calculate returns.
+    if lag:
+        returns = prices.pct_change(1).shift(-1)
+    else:
+        returns = prices.pct_change(1)
 
```

Comment:
Please, let the user define the period (for both simple and log returns).

Reply:
See the reply in the excess median PR. Happy with this change.